server/controllers/contest.js

```diff
@@ -65,4 +69,42 @@ exports.getAllContestsByUser = asyncHandler(async (req, res) => {
   } catch (err) {
     res.status(500).json(err);
   }
+})
+
+exports.getContestsByDeadlineDate = asyncHandler(async (req, res) => {
```

Comment:
Maybe use the `getAllContests` controller to implement this logic? By doing so, we will only have one route and one controller 👍🏽

Reply:
a little unclear what you mean here, one controller requires a query parameter and one does not.

Reply:
You can add the query parameters to the `getAllContests` controller. If the parameters have not been specified (i.e undefined), you can fetch all the contests. On the other hand, if paraments have been specified, you can fetch contests based on the date range.

Note: On the FE, we may need to have a button to reset the date filters.

Reply:
sounds good, will fix this after work today